src/main/kotlin/org/rust/ide/inspections/checkMatch/RsNonExhaustiveMatchInspection.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.checkMatch
+
+import org.rust.ide.inspections.RsLocalInspectionTool
+import org.rust.ide.inspections.RsProblemsHolder
+import org.rust.lang.core.psi.RsMatchExpr
+import org.rust.lang.core.psi.RsVisitor
+import org.rust.lang.core.psi.ext.arms
+import org.rust.lang.core.types.infer.containsTyOfClass
+import org.rust.lang.core.types.ty.TyUnknown
+import org.rust.lang.core.types.type
+import org.rust.lang.utils.RsDiagnostic
+import org.rust.lang.utils.addToHolder
+
+class RsNonExhaustiveMatchInspection : RsLocalInspectionTool() {
+    override fun getDisplayName() = "Non-exhaustive match"
+
+    override fun buildVisitor(holder: RsProblemsHolder, isOnTheFly: Boolean) = object : RsVisitor() {
+        override fun visitMatchExpr(matchExpr: RsMatchExpr) {
+            val exprType = matchExpr.expr?.type ?: return
+            if (exprType.containsTyOfClass(TyUnknown::class.java)) return
+            try {
+                checkExhaustive(matchExpr, holder)
+            } catch (todo: NotImplementedError) {
+            } catch (e: CheckMatchException) {
+            }
+        }
+    }
+
+    private fun checkExhaustive(match: RsMatchExpr, holder: RsProblemsHolder) {
```

Comment:
```suggestion
    private fun checkExhaustiveness(match: RsMatchExpr, holder: RsProblemsHolder) {
```

Reply:
This name was given to match the original `check_exhaustive` method in rustc implementation

Reply:
ok